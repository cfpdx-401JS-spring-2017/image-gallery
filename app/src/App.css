.App {
  text-align: center;
  display: flex;
  height: 100vh;
  width: 100%;
  justify-content: center;
  align-items: center;

}

.ThumbNail {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  height: 30vh;
}

.List {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
}

button {
  height: 10vh;
  width: 20vw;
  background-color: #476A6F;
  border: none;
  margin: 2vh 2vh;
  box-shadow: 1px 1px 1px 1px #202A25;
  border-radius: 10px;
}

.Link {
  color: #F7F0F0;
  text-decoration: none;
  font-size: 2em;
}

.Link:hover {
  
}
