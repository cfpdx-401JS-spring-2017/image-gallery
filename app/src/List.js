import React, { Component } from 'react';
import PropTypes from 'prop-types';

class List extends Component {

  static propTypes = {
    data: PropTypes.array.isRequired
  }

  render() {
    const { data } = this.props;

    return (
      <div>
        {data.map((img, i) => (
          <ul key={i}>
            <li>
              {img.title}
            </li>
            <li>
              <a href=''>
                {img.url}
              </a>
            </li>
            <li>
              {img.description}
            </li>
          </ul>
        ))}
      </div>
    );
  }
}

export default List;