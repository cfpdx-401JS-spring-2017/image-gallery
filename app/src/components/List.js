import React, { Component } from 'react';
import PropTypes from 'prop-types';

class List extends Component {

  static propTypes = {
    data: PropTypes.array.isRequired
  }

  render() {
    const { data, onDelete } = this.props;

    return (
      <div>
        {data.map((img) => (
          <ul key={img._id}>
            <li className="list-title">
              {img.title}
            </li>
            <li>
              <a href="">
                {img.url}
              </a>
            </li>
            <li>
              {img.description}
            </li>
            <li>
              <button onClick={() => onDelete(img._id)}
                      className="delete-button"
              >
                Delete ðŸ—‘
              </button>
            </li>
          </ul>
        ))}
      </div>
    );
  }
}

export default List;