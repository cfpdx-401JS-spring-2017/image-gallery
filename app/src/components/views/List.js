import React from 'react';
import PropTypes from 'prop-types';
import ListImage from './ListImage';

List.propTypes = {
  poms: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    description: PropTypes.string,
    url: PropTypes.string.isRequired
  }))
};

export default function List({ poms, handleDelete }) {
  return (
    <ul>
      {poms.map((pom, i) => <ListImage
        key={pom._id}
        {...pom}
        onDelete={() => handleDelete(pom._id, i)}
      />)}
    </ul>
  );
}