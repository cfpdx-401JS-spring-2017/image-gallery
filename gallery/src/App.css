.App {
  text-align: center;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}

.ThumbNail {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  height: 30vh;
}

.List {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
}

button {
  height: 8vh;
  width: 12vw;
  background-color: #2F3E46;
  color:#CAD2C5;
  border: none;
  margin: 2vh 2vh;
  box-shadow: none;
}
