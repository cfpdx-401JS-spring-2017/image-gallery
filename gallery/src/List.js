import React, { Component } from 'react';

class List extends Component {
  constructor() {
    super();
    this.state ={

    };
  }
  

  render() {
    return (
      <div className="List">
        {this.props.images.map((image, i) =>{
            let {title, description, url} = image;
          return (
            <div key={i} style={{ width: '30vw', margin: '5px 5px'}}>
              <h3>{title}</h3> 
              <p>{description}</p> 
              <p>{url}</p>
            </div>
          );
          
        })}
      </div>  
    );
  }
}

export default List;