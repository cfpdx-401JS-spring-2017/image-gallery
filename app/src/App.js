import React, { Component } from 'react';
import {
  BrowserRouter as Router,
  Route,
  Switch,
  Redirect
} from 'react-router-dom';

import Home from './components/Home';
import Albums from './components/albums/Albums';
import './styles/App.css';

class App extends Component {
  render() {
    return (
      <Router>
        <div className="App">
          <h2>CPDB: Cute Pomeranian Database</h2>
          <p>
            Welcome to the <code>Cute Pomeranian Database</code>! Check out our poms by clicking below:
          </p>
          <Switch>
            <Route exact path='/' component={Home}/>
            <Route path='/albums' component={Albums} />
            <Redirect to='/' />
          </Switch>
        </div>
      </Router>
    );
  }
}

export default App;
