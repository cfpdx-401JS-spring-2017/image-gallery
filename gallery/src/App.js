import React, { Component } from 'react';
import './App.css';
import AlbumsViewer from './AlbumsViewer';
import Albums from './Albums';

import {
  BrowserRouter as Router,
  Route, Switch, Redirect, Link
} from 'react-router-dom'

class App extends Component {
  constructor() {
    super();
    this.state = {
    }

  }

  render() {
  
    return (
      <div className="App">
        <Router>
            <div>
                <Switch>
                    {/*<Route exact path="/" component={App}/>*/}
                    <Route path="/albums/:id" component={AlbumsViewer}/>
                    <Route path="/albums" component={Albums}/>
                    <Redirect path="/albums" component={Albums}/>
                </Switch>
                <button>
                  <Link to="/albums">Go to Albums</Link>
                </button>
            </div>
        </Router>
      </div>  
    );
  }
}

export default App;
