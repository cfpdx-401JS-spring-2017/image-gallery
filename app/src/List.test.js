import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import List from './List';

import renderer from 'react-test-renderer';

const images = [ { 
    title: 'Cool Waterfall',
    description: 'This waterfall is located in heart of Africa near the mysterious Triad Canyon',
    url: 'ImageOne',
    _id: 1
    }
]
it('renders without crashing', () => {
    
  const div = document.createElement('div');
  ReactDOM.render(<List images={images}/>, div);
});

it('renders a snapshot of main List component', () => {
  const component = renderer.create(<List images={images} />);
  const bobo = component.toJSON();
  expect(bobo).toMatchSnapshot();
});