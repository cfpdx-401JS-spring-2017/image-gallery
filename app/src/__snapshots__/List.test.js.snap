// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot of main List component 1`] = `
<div
  className="List"
>
  <div
    style={
      Object {
        "backgroundColor": "#84A98C",
        "color": "#354F52",
        "margin": "5vh 5vh",
        "width": "25vw",
      }
    }
  >
    <h3>
      Cool Waterfall
    </h3>
    <p>
      This waterfall is located in heart of Africa near the mysterious Triad Canyon
    </p>
    <p>
      ImageOne
    </p>
    <button
      onClick={[Function]}
      value={1}
    >
      delete
    </button>
  </div>
</div>
`;
